yt/yt/server/master/security_server/security_manager.cpp

```diff
@@ -3791,6 +3801,11 @@ class TSecurityManager
             DoAddMember(UsersGroup_, SuperusersGroup_);
         }
 
+        // admins
+        if (EnsureBuiltinGroupInitialized(AdminsGroup_, AdminsGroupId_, AdminsGroupName)) {
```

Comment:
I see some danger here: on many clusters admin group is already created, but has different id, so we'll try to create group and creation will fail because of the duplicate name. Let's make `EnsureBuiltinGroupInitialized` better to fix it.

Reply:
`EnsureBuiltinGroupInitialized` is already good enough and returns false if the group name or id is occupied: https://github.com/ytsaurus/ytsaurus/blob/b84545f70c8ece59142d2beedbcd13232e65da1e/yt/yt/server/master/security_server/security_manager.cpp#L3921-L3927

Reply:
Is it?

https://github.com/ytsaurus/ytsaurus/blob/b84545f70c8ece59142d2beedbcd13232e65da1e/yt/yt/server/master/security_server/security_manager.cpp#L2985

Here we fail in case if name is already occupited. And seems like there is no check that group with given name (not id) already exists in `EnsureBuiltinGroupInitialized`.

Reply:
Seems like you are right, will try to fix it

Reply:
Fixed

Reply:
Is it ok that `AdminGroup->GetId() != AdminGroupId_` in this case? I'd prefer to overwrite `AdminGroupId_` with actual id in this case.

Reply:
I don't think it matters now, since no *GroupId_ variables are actually used in code. But it may change in future

I adjusted `EnsureBuiltinGroupInitialized` to overwrite groupId in case it found the group by name